dmscripts/send_dos_opportunities_email.py

```diff
@@ -1,23 +1,22 @@
 from datetime import datetime, date, timedelta
 
-from requests.exceptions import RequestException
-
 from dmscripts.helpers.html_helpers import render_html
 from dmscripts.helpers import logging_helpers
 from dmscripts.helpers.logging_helpers import logging
 from dmutils.formats import DATETIME_FORMAT, DISPLAY_DATE_FORMAT
 
-import dmapiclient
 
 logger = logging_helpers.configure_logger({'dmapiclient': logging.INFO})
 
 
 def get_live_briefs_between_two_dates(data_api_client, lot_slug, start_date, end_date):
-    return [
-        brief for brief in data_api_client.find_briefs_iter(status="live", lot=lot_slug, human=True)
-        if datetime.strptime(brief['publishedAt'], DATETIME_FORMAT).date() >= start_date
-        and datetime.strptime(brief['publishedAt'], DATETIME_FORMAT).date() <= end_date
-    ]
+    """Get all briefs for a lot which were published between 2 dates."""
+    valid_briefs = []
+    for brief in data_api_client.find_briefs_iter(status="live", lot=lot_slug, human=True):
+        brief_published_at = datetime.strptime(brief['publishedAt'], DATETIME_FORMAT).date()
+        if start_date <= brief_published_at <= end_date:
+            valid_briefs.append(brief)
+    return valid_briefs
```

Comment:
...and if it had complained about the list comprehension's complexity measure and not the manual loop that would have been damning evidence against it, because there are far *far* more ways the manual loop can be subtly wrong.